### PR TITLE
Add missing SCC for OTEL sts for OCP < 4.12

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

This PR fixes issue https://dt-rnd.atlassian.net/browse/DAQ-2995

Steps to reproduce of bug:

Steps to reproduce:
1. Deploy operator release-1.4
2. apply DK with extensions enabled
3. describe (OTEL) collector statefulset:

```
│ Events:                                                                                                                                      │
│   Type     Reason        Age                   From                    Message                                                               │
│   ----     ------        ----                  ----                    -------                                                               │
│   Warning  FailedCreate  84s (x18 over 7m19s)  statefulset-controller  create Pod dynakube-extensions-collector-0 in StatefulSet dynakube-ex │
│ tensions-collector failed error: pods "dynakube-extensions-collector-0" is forbidden: unable to validate against any security context constr │
│ aint: [pod.metadata.annotations.seccomp.security.alpha.kubernetes.io/pod: Forbidden: seccomp may not be set pod.metadata.annotations.contain │
│ er.seccomp.security.alpha.kubernetes.io/collector: Forbidden: seccomp may not be set provider "anyuid": Forbidden: not usable by user or ser │
│ viceaccount provider "nonroot": Forbidden: not usable by user or serviceaccount provider "hostmount-anyuid": Forbidden: not usable by user o │
│ r serviceaccount provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount provider "hostnetwork": Forbidd │
│ en: not usable by user or serviceaccount provider "hostaccess": Forbidden: not usable by user or serviceaccount pod.metadata.annotations.sec │
│ comp.security.alpha.kubernetes.io/pod: Forbidden: seccomp may not be set pod.metadata.annotations.container.seccomp.security.alpha.kubernete │
│ s.io/collector: Forbidden: seccomp may not be set provider "node-exporter": Forbidden: not usable by user or serviceaccount provider "privil │
│ eged": Forbidden: not usable by user or serviceaccount]   
Found root cause of failing e2e tests on
```

## How can this be tested?

Run `make test/e2e/supportarchive`

since this test-case will install dk (with extensions) and edgeconnect.

repeat test for OCP-4.10, OCP-4.11, OCP-4.14 ;) have fun!

### CODE REVIEW

File: config/helm/chart/default/templates/Common/extensions-opentelemetry-collector/clusterrole-extensions-opentelemetry-collector.yaml [hunk 1]
```diff
@@ -44,6 +44,17 @@ rules:
       - get
       - list
       - watch
+  {{- if (eq (include "dynatrace-operator.openshiftOrOlm" .) "true") }}
+  - apiGroups:
+      - security.openshift.io
+    resourceNames:
+      - privileged
+      - nonroot-v2
```
Comment (0sewa0) [inline config/helm/chart/default/templates/Common/extensions-opentelemetry-collector/clusterrole-extensions-opentelemetry-collector.yaml:52]: you don't need `nonroot-v2` as you don't set a custom `runAs` in the securityContext of the otel `StatefulSet`
(`restricted-v2` is enough in OCP 4.12 and later, and you don't need a role to use that)

Comment (andriisoldatenko) [reply config/helm/chart/default/templates/Common/extensions-opentelemetry-collector/clusterrole-extensions-opentelemetry-collector.yaml]: removed in dc0115b43

File: pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
```
Comment (0sewa0) [inline pvc.yaml:2]: ?

File: sts.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
+apiVersion: apps/v1
```
Comment (0sewa0) [inline sts.yaml:1]: ?

Comment (andriisoldatenko) [reply sts.yaml:1]: lol

Comment (andriisoldatenko) [reply sts.yaml:1]: advanced git skills

**Review by 0sewa0 — COMMENTED** [review]
maybe unit-test 👀

**Comment by andriisoldatenko** [comment]
> maybe unit-test 👀

helm unit test you mean? on it

**Comment by andriisoldatenko** [comment]
@0sewa0 test has been added in 98f555131

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**